src/search/main.cpp

```diff
@@ -19,10 +19,15 @@
 ===========================================================================
 */
 
-#include "search_server.h"
+#include "search_application.h"
+#include "search_engine.h"
 
 int main(int argc, char** argv)
 {
-    std::make_unique<SearchServer>(argc, argv)->run();
+    const auto searchApp = std::make_unique<SearchApplication>(argc, argv);
+
+    searchApp->initialize();
```

Comment:
I know you're refactoring ahead of setting up xi_test, but this two-phase initialisation is pretty fragile, do we really need it?